lib/npo.src.js

```diff
@@ -77,6 +124,15 @@
 			cycle = timer(scheduling_queue.drain);
 		}
 	}
+	
+	function hasCatch(chain) {
+		for (var i = 0; i < chain.length; i++) {
+			if (chain[i].failure) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
Note there is a version of forked NPO used in core-js I believe that implements these events - this implementation should probably pass their tests (hopefully they have some) or Node's for native promises before merging.